custom/conf/app.ini.sample

```diff
@@ -792,3 +792,9 @@ IS_INPUT_FILE = false
 ENABLED = false
 ; If you want to add authorization, specify a token here
 TOKEN =
+
+[storage]
+; URL of Bucket where files, such as, attachments, avatars, etc. are stored
+; If unset, file://<current_directory> is used to maintain backward compatibility
```

Comment:
"`backward compatibility`" makes it sound like buckets are the "modern" way to go. Could this be changed to something like "`if unset, {APP_DATA_PATH}/attachments is used`" (not checked) or something around those lines?

Reply:
Actually if unset, it now uses `AppWorkPath`. When using this path for `attachments` or `avatars`, `AttachmentPath` or `AvatarUploadPath` is appended to this `BucketURL`.
If `AttachmentPath` is unset, we set this as `data/attachments`, to maintain similarity between all the available buckets. https://github.com/go-gitea/gitea/blob/83ccf4c649e58abb83c7d21ec42a3f378be51949/modules/setting/setting.go#L803

Reply:
So, if I understand well, the user can't use `APP_DATA_PATH` anymore? I think it's too big of a breaking change. If the user upgrades because they were interested in X other functionality the new version provides, he or she might not be aware of this change and get in a lot of trouble. Can't they both co-exist? e.g. if there are both `APP_DATA_PATH` and `BUCKET_URL`, then the former should be ignored, but if there's only `APP_DATA_PATH`, the old behaviour should be honored.

Reply:
`APP_DATA_PATH` related issue is now fixed. 
https://github.com/go-gitea/gitea/blob/4c390c9699ee6400a9467c4c8234af9efba5acaf/modules/setting/setting.go#L1053-L1083

Now, `APP_DATA_PATH`and `BUCKET_URL` both co-exist. Please let me know if I did it right.